### PR TITLE
releases, toc: add v4.0.0-beta.2 Release Notes

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add TiDB v4.0.0-beta.2 Release Notes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: releases/4.0.0-beta.2.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
++ TiDB Ansible
+
+## Bug 修复
+
++ TiDB
+
++ TiKV
+    - 修复 backup 在遇到空的 short value 时可能 panic 的问题 [#6718](https://github.com/tikv/tikv/pull/6718)
+    - 修复 hibernate region 在某些特殊条件下未被正确唤醒的问题 [#6772](https://github.com/tikv/tikv/pull/6672) [#6648](https://github.com/tikv/tikv/pull/6648) [#6376](https://github.com/tikv/tikv/pull/6736)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]:
```suggestion
    - 修复 Hibernate Region 在某些特殊条件下未被正确唤醒的问题 [#6772](https://github.com/tikv/tikv/pull/6672) [#6648](https://github.com/tikv/tikv/pull/6648) [#6376](https://github.com/tikv/tikv/pull/6736)
```

File: releases/4.0.0-beta.2.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
++ Tools
+    - TiDB Binlog
+        - 对 Drainer 的配置项 `disable-dispatch` 和 `disable-causality` 做了兼容 [#915](https://github.com/pingcap/tidb-binlog/pull/915)
+
+## 新功能
+
++ TiDB
+
++ TiKV
+    - 支持将动态修改配置的结果持久化存储到硬盘 [#6684](https://github.com/tikv/tikv/pull/6684)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]:
```suggestion
    - 支持将动态修改配置的结果持久化存储到硬盘 [#6684](https://github.com/tikv/tikv/pull/6684)
```

File: releases/4.0.0-beta.2.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+## Bug 修复
+
++ TiDB
+
++ TiKV
+    - 修复 backup 在遇到空的 short value 时可能 panic 的问题 [#6718](https://github.com/tikv/tikv/pull/6718)
+    - 修复 hibernate region 在某些特殊条件下未被正确唤醒的问题 [#6772](https://github.com/tikv/tikv/pull/6672) [#6648](https://github.com/tikv/tikv/pull/6648) [#6376](https://github.com/tikv/tikv/pull/6736) 
+
++ PD
+    - 修复因 rule checker 在给 region 分配 Store 失败导致系统 Panic 的问题 [#2160](https://github.com/pingcap/pd/pull/2160)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]:
```suggestion
    - 修复因 rule checker 在给 Region 分配 store 失败导致系统 panic 的问题 [#2160](https://github.com/pingcap/pd/pull/2160)
```

File: releases/4.0.0-beta.2.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+TiDB 版本：4.0.0-beta.2
+
+TiDB Ansible 版本：4.0.0-beta.2
+
+## TiDB 4.0.0-beta.2 兼容性变化
+
++ Tools
+    - TiDB Binlog
+        - 对 Drainer 的配置项 `disable-dispatch` 和 `disable-causality` 做了兼容 [#915](https://github.com/pingcap/tidb-binlog/pull/915)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]: Please check the updated version in the Google Docs and update this note accordingly.

File: releases/4.0.0-beta.2.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+TiDB 版本：4.0.0-beta.2
+
+TiDB Ansible 版本：4.0.0-beta.2
+
+## TiDB 4.0.0-beta.2 兼容性变化
+
++ Tools
+    - TiDB Binlog
+        - 修复 Drainer 配置`disable-dispatch`、`disable-causality` 时系统直接报错并退出的问题 [#915](https://github.com/pingcap/tidb-binlog/pull/915)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]:
```suggestion
        - 修复 Drainer 配置 `disable-dispatch`、`disable-causality` 时系统直接报错并退出的问题 [#915](https://github.com/pingcap/tidb-binlog/pull/915)
```

File: releases/4.0.0-beta.2.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - 修复 Hibernate Region 在某些特殊条件下未被正确唤醒的问题 [#6772](https://github.com/tikv/tikv/pull/6672) [#6648](https://github.com/tikv/tikv/pull/6648) [#6376](https://github.com/tikv/tikv/pull/6736)
+
++ PD
+    - 修复因 rule checker 在给 Region 分配 store 失败导致系统 panic 的问题 [#2160](https://github.com/pingcap/pd/pull/2160)
+    - 修复启用动态修改配置功能后，配置可能在切换 leader 时有同步延迟的问题 [#2154](https://github.com/pingcap/pd/pull/2154) 
+
++ Tools
+    - BR
+        - 修复因 PD 无法处理过大消息导致在数据规模较大时恢复失败的问题 [#167](https://github.com/pingcap/br/pull/167)
+        - 修复因 BR 与 TiDB 版本不兼容导致升级 TiDB 的问题 [#186](https://github.com/pingcap/br/pull/186)
```
Comment (TomShawn) [inline releases/4.0.0-beta.2.md]:
```suggestion
        - 修复因 BR 与 TiDB 版本不兼容导致 BR 运行失败的问题 [#186](https://github.com/pingcap/br/pull/186)
```

File: releases/4.0.0-beta.2.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
++ PD
+    - 支持将动态修改配置的结果持久化存储到硬盘 [#2153](https://github.com/pingcap/pd/pull/2153)
+
++ Tools
+    - TiDB Binlog
+        - 新增 TiDB 集群之间数据双向复制功能 [#879](https://github.com/pingcap/tidb-binlog/pull/879) [#903](https://github.com/pingcap/tidb-binlog/pull/903)
+    - TiDB Lightning
+        - 新增配置 TLS 功能 [#40](https://github.com/tikv/importer/pull/40) [#270](https://github.com/pingcap/tidb-lightning/pull/270)
+    - 新增 TiCDC 工具，用于捕捉 TiKV 变化的数据并同步到下游 Kafka、MySQL 协议的数据库；确保数据最终一致性，若下游是 Kafka 可确保行级别的有序；提供进程级别的高可用能力。
```
Comment (ran-huang) [inline releases/4.0.0-beta.2.md]: @TomShawn Shall we separate this sentence into several sub-items, just like the English version?
```suggestion
    - 新增 TiCDC 工具，提供以下功能：
        - 捕捉 TiKV 变化的数据，同步到下游 Kafka、MySQL 协议的数据库
        - 确保数据最终一致性，若下游是 Kafka，也可确保行级别的有序
        - 提供进程级别的高可用能力
```

Comment (TomShawn) [reply releases/4.0.0-beta.2.md]: I think this is a better way, but you can check with the PM.

Comment (ran-huang) [reply releases/4.0.0-beta.2.md]: @scsldb PTAL~

File: releases/4.0.0-beta.2.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+---
+title: TiDB 4.0.0-beta.2 Release Notes
```
Comment (ran-huang) [inline releases/4.0.0-beta.2.md]:
```suggestion
title: TiDB 4.0.0 Beta.2 Release Notes
```

File: releases/4.0.0-beta.2.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+category: Releases
+---
+
+# TiDB 4.0.0-beta.2 Release Notes
```
Comment (ran-huang) [inline releases/4.0.0-beta.2.md]:
```suggestion
# TiDB 4.0.0 Beta.2 Release Notes
```

**Comment by scsldb** [comment]
LGTM

**Pull request merged**